.github/workflows/check-build.yml

```diff
@@ -0,0 +1,269 @@
+      updated_specs: ${{ needs.detect-changes.outputs.updated_specs }}
+      updated_extended_specs: ${{ needs.detect-changes.outputs.updated_extended_specs }}
+      azl_extra_make_args: ${{ needs.detect-changes.outputs.azl_extra_make_args }}
+
+    outputs:
+      daily_build_id: ${{ steps.find-daily-repo.outputs.DAILY_BUILD_ID }}
+      daily_build_arch: ${{ steps.find-daily-repo.outputs.DAILY_BUILD_ARCH }}
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@v4
+
+      - name: Setup Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+
+      - name: Install Python dependencies for tools
+        run: python3 -m pip install -r toolkit/scripts/requirements.txt
+
+      - name: Find daily repo
+        id: find-daily-repo
+        run: |
+          DAILY_BUILD_ID=$(toolkit/scripts/get_lkg_id.sh /tmp)
+          echo "DAILY_BUILD_ID=${DAILY_BUILD_ID}" >>$GITHUB_OUTPUT
+          echo "DAILY_BUILD_ARCH=x86-64" >>$GITHUB_OUTPUT
+
+      - name: "Build modified specs (base)"
+        id: build-specs
+        if: ${{ env.updated_specs != '' }}
+        run: |
+          sudo make -C toolkit \
+            build-packages \
+            $azl_extra_make_args \
+            -j$(nproc) \
+            REBUILD_TOOLS=y \
+            SRPM_PACK_LIST='${{ env.updated_specs }}'
+
+      - name: "Build modified specs (extended)"
+        id: build-extended-specs
+        if: ${{ env.updated_extended_specs != '' }}
+        run: |
+          sudo make -C toolkit \
+            build-packages \
+            $azl_extra_make_args \
+            -j$(nproc) \
+            REBUILD_TOOLS=y \
+            SPECS_DIR=$(pwd)/SPECS-EXTENDED \
+            SRPM_PACK_LIST='${{ env.updated_extended_specs }}'
+
+      - name: Upload built RPMs
+        uses: actions/upload-artifact@v4
+        with:
+          name: rpms
+          path: out/RPMS
+
+      - name: Upload build logs
+        id: upload-build-logs
+        if: always()
+        uses: actions/upload-artifact@v4
+        with:
+          name: build-logs-${{ github.run_attempt }}
+          path: build/logs
+
+      - name: Publish result
+        if: always()
+        run: |
+          if [[ ( ${{ steps.build-specs.outcome }} == 'success' || ${{ steps.build-specs.outcome }} == 'skipped' ) && ( ${{ steps.build-extended-specs.outcome }} == 'success' || ${{ steps.build-extended-specs.outcome }} == 'skipped' ) ]]; then
+            echo "✅ Build succeeded" >>${GITHUB_STEP_SUMMARY}
+          else
+            echo "❌ Build failed" >>${GITHUB_STEP_SUMMARY}
+          fi
+
+      - name: Advertise build logs
+        if: always() && steps.upload-build-logs.outcome == 'success'
+        run: |
+          echo "_Build logs available [for download](${{ steps.upload-build-logs.outputs.artifact-url }})_" >>${GITHUB_STEP_SUMMARY}
+
+  install:
+    name: 'Test installing RPM'
+    needs: [build, detect-changes]
```

Comment:
Do we need to add detect-changes here since build already depends on detect-changes ?

Reply:
It will be needed in my opinion for line number 217 since we need the output from that job, which is not available from build for updated_specs/updated_extended_specs.